Source/WebCore/loader/cache/CachedResourceLoader.cpp

```diff
@@ -498,44 +501,47 @@ bool CachedResourceLoader::allowedByContentSecurityPolicy(CachedResource::Type t
 
     ASSERT(m_document);
     ASSERT(m_document->contentSecurityPolicy());
+    auto contentSecurityPolicy = m_document->contentSecurityPolicy();
+    if (!contentSecurityPolicy)
+        return true;
```

Comment:
This will still crash in Release/Production builds if `m_document` is `nullptr`.  We should add a `nullptr` check, not just assert.

This will still crash in Release/Production builds if `m_document` is `nullptr`.  We should add a `nullptr` check, not just assert.  Maybe:

```
    ASSERT(m_document);
    if (!m_document)
        return false;  // Or true???
```

Reply:
I initially had that, but realized that all calls to this method already nullptr check document. But I can add it in case a new call site is added that does not do the check.

Reply:
I think we return true; if no document, then no security origin, therefore no security boundary to check against, therefore no violation.